src/DoctrineORMModule/Service/EntityResolverFactory.php

```diff
@@ -42,7 +43,11 @@ public function createService(ServiceLocatorInterface $serviceLocator)
             $targetEntityListener->addResolveTargetEntity($oldEntity, $newEntity, array());
         }
 
-        $eventManager->addEventListener(Events::loadClassMetadata, $targetEntityListener);
+        if (Version::compare('2.5.0') >= 0) {
```

Comment:
@bakura10 instead of this, why not checking if `$targetEntityListener` implements the subscriber interface?